lgc/disassembler/Disassembler.cpp

```diff
@@ -64,6 +65,62 @@ const char *getPalMetadataRegName(unsigned regNumber);
 
 namespace {
 
+struct InstOp {
+  Optional<int64_t> imm;
+  Optional<unsigned> sReg;
+};
+
+struct InstOrDirective {
```

Comment:
Could you add some documentation explaining what these `struct`s are for and why we need them?